app/ledger.js

```diff
@@ -441,9 +448,24 @@ var initialize = (onoff) => {
         }
 
         getStateInfo(state)
+
         try {
-          client = ledgerClient(state.personaId,
-                                              underscore.extend(state.options, { roundtrip: roundtrip }, clientOptions), state)
+          client = ledgerClient(
+            state.personaId,
+            underscore.extend(state.options, { roundtrip: roundtrip }, clientOptions),
+            state
+          )
+
+          // Scenario: User enables Payments, disables it, waits 30+ days, then
+          // enables it again -> reconcileStamp is in the past.
+          // In this case reset reconcileStamp to the future.
+          if (reason === 'changeSettingPaymentsEnabled') {
+            let timeUntilReconcile = client.timeUntilReconcile()
+            if (typeof timeUntilReconcile === 'number' && timeUntilReconcile < 0) {
+              let futureReconcileStamp = underscore.now() + 30 * msecs.day
```

Comment:
it is better to call `setTimeUntilReconcile` with a `null` timestamp and let the `ledger-client` package figure out the next time.